### DISCUSSION TITLE
Unable to start keycloak when db password is passed as command line argument which contains space (--db-password=filepwd userpwd)

### DISCUSSION THREAD

**Discussion by ramya2428** [keycloak/keycloak#26404] [Archive]
Hi,

I am using latest version of keycloak 23.0.4 with h2 db. My requirement is I need to encrypt the DB. So after looking after many blogs I found a way to encrypt using AES algorithm. Hence I am passing db password as command line argument so that i can encrypt and send like --db-password=filepwd userpassword. As there is space in the password i am getting below error
  Option '--db-password' expects a single value (password)
  Try 'kc.sh start --help' for more information on the available options.

Also if i give complete url in keycloak.conf file it works as expected, like
db-url=jdbc:h2:file:KEYCLOAK_PATH/data/h2/keycloakdb;AUTO_SERVER=TRUE;NON_KEYWORDS=VALUE;CIPHER=AES;user=sa;password=filepwd userpassword

Is this a known bug on keycloak?  Because similar issue is found in 
https://github.com/keycloak/keycloak/pull/22585#issuecomment-1693578357

Any help to know how to deal with space in the password would be helpful

Thanks

**Comment by danielFesenmeyer** [comment]
I think it's not a bug Keycloak - from my understanding, it's just how whitespaces are handled in bash and other shells. Probably it's best just to put your whole parameters string into double quotes.

See [here](https://unix.stackexchange.com/questions/131766/why-does-my-shell-script-choke-on-whitespace-or-other-special-characters) for more information.
